src/tasks.ts

```diff
@@ -1,87 +1,140 @@
 'use strict';
 
 import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as path from 'path';
 
 interface RTaskDefinition extends vscode.TaskDefinition {
-    command: string;
-    args: string[];
+    code: string[];
     cwd?: string;
     env?: { [key: string]: string };
 }
 
-export class RTaskProvider implements vscode.TaskProvider {
-    public readonly type = 'R';
 
-    private getTask(name: string, definition: RTaskDefinition, problemMatchers?: string | string[]): vscode.Task {
-        return new vscode.Task(
-            definition,
-            vscode.TaskScope.Workspace,
-            name,
-            'R',
-            new vscode.ProcessExecution(
-                definition.command,
-                definition.args,
-                {
-                    cwd: definition.cwd,
-                    env: definition.env
-                }
-            ),
-            problemMatchers
-        );
+const TYPE = 'R';
+
+class TaskMeta {
+    constructor(
+        public name: string,
+        public definition: RTaskDefinition,
+        public group?: vscode.TaskGroup,
+        public problemMatchers?: string | string[],
+    ) {}
+}
+
+
+const rtasks: TaskMeta[] = [
+    new TaskMeta(
+        'Test',
+        {
+            type: TYPE,
+            code: ['devtools::test()']
+        },
+        vscode.TaskGroup.Test,
+        '$testthat'
+    ),
+    
+    new TaskMeta(
+        'Build',
+        {
+            type: TYPE,
+            code: [ 'devtools::build()' ]
+        },
+        vscode.TaskGroup.Build
+    ),
+
+    new TaskMeta(
+        'Check',
+        {
+            type: TYPE,
+            code: [ 'devtools::check()' ]
+        },
+        vscode.TaskGroup.Test
+    ),
+    
+    new TaskMeta(
+        'Document',
+        {
+            type: TYPE,
+            code: [ 'devtools::document()' ]
+        },
+        vscode.TaskGroup.Build
+    ),
+    
+    new TaskMeta(
+        'Install',
+        {
+            type: TYPE,
+            code: [ 'devtools::install()' ]
+        },
+        vscode.TaskGroup.Build
+    )
+];
+
+
+const asTask = function (
+    folder: vscode.WorkspaceFolder | vscode.TaskScope,
+    meta: TaskMeta
+): vscode.Task {
+    const task: vscode.Task = new vscode.Task(
+        meta.definition,
+        folder,
+        meta.name,
+        TYPE,
+        new vscode.ProcessExecution(
+            'Rscript',
+            ['-e', meta.definition.code.join('; ')],
```

Comment:
Is there a difference between supplying `-e joined-expression` and supplying multiple `-e expr` such as `Rscript -e expr1 -e expr2`.

Reply:
I don't think there's any difference but switched it to multiple `-e`'s as I imagine that's probably safer than injecting line breaks.